Improve List demo description - list selection (#560)
DevExpress/devextreme-demos: JSDemos/Demos/List/ListSelection/description.md

> The **List** offers four selection modes: *none*, *all*, *single*, and *multi*. 
> 
> Selection is disabled by default. To enable it, pass one of the mentioned above values to the [selectionMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#selectionMode) property.

Reviewer: - "none" isn't really a selection mode, since it disables item selection.
- Making the entire description about the demo is bad, but so is not referring to the demo at all.

```suggestion
The **List** offers three item selection modes: *"single"*, *"multiple"*, and *"all"*.  To enable item selection, pass a mode name to the [selectionMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#selectionMode) property. If you set this property to *"none"* (default setting), item selection is disabled. In this example, you can try out all the selection modes. Use the Selection Mode drop-down menu to switch between them.
```

> Users can click or tap on the **List** item to select it. Selected items become shaded. It is possible to add selection checkboxes or radio buttons to list items. To do so, set the [showSelectionControls](/Documentation/ApiReference/UI_Components/dxList/Configuration/#showSelectionControls) property to **true**. This property also must be set to *true* when you use the *all* selection mode.

Reviewer: Since in this demo showSelectionControls is enabled, we should start the description from the situation when the controls are already visible.

```suggestion
List items can display checkboxes or radio buttons that help users select the items. To show these controls, enable the [showSelectionControls](/Documentation/ApiReference/UI_Components/dxList/Configuration/#showSelectionControls) property. Otherwise, users click or tap list items to select them. The **showSelectionControls** property must also be enabled when you use the *"all"* **selectionMode** so that the Select All checkbox is visible.
```

> When the **List** data is [paginated](/Documentation/Guide/UI_Components/List/Paging/), you can choose whether the "Select All" check box will select all items on all pages or only the items on the currently rendered pages. To configure this functionality, set the [selectAllMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#selectAllMode) property to *allPAges* or *page*.

Reviewer: Let's not forget about the functionality of this demo.

```suggestion
When the **List** data is [paginated](/Documentation/Guide/UI_Components/List/Paging/), you can choose whether the Select All checkbox will select all items on all pages or only the items on the currently rendered pages. To configure this functionality, set the [selectAllMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#selectAllMode) property to *"allPages"* or *"page"*. In this demo, you can use the Select All Mode drop-down menu to change the **selectAllMode** property at runtime.
```

> To configure the initial selection, use the [selectedItemKeys](/Documentation/ApiReference/UI_Components/dxList/Configuration/#selectedItemKeys) property. The property accepts a array of item keys wrapped in quotation marks.

Reviewer: ```suggestion
To configure the initial selection or access the keys of currently selected items, use the [selectedItemKeys](/Documentation/ApiReference/UI_Components/dxList/Configuration/#selectedItemKeys) array. In this example, you can see the contents of this array in the Selected IDs section below the List.
```

> You can use the [onSelectionChanged](/Documentation/ApiReference/UI_Components/dxList/Configuration/#onSelectionChanged) property to perform an action when users select an item or items. In this demo, this property is used to display IDs of selected items.

Reviewer: Angular, Vue, and React don't use `onSelectionChanged`. Let's use a framework-neutral text.

```suggestion
You can also implement the [onSelectionChanged](/Documentation/ApiReference/UI_Components/dxList/Configuration/#onSelectionChanged) event handler to perform an action when users select an item or items.
```

> The **List** offers three item selection modes: *"single"*, *"multiple"*, and *"all"*.  To enable item selection, pass a mode name to the [selectionMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#selectionMode) property. If you set this property to *"none"* (default setting), item selection is disabled. In this example, you can try out all the selection modes. Use the Selection Mode drop-down menu to switch between them.

Reviewer 2: ```suggestion
There are three **List** item selection modes: *"single"*, *"multiple"*, and *"all"*.  To enable item selection, pass the mode name to the [selectionMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#selectionMode) property. If you set this property to *"none"* (default setting), item selection is disabled. In this demo, you can try out all the selection modes. Use the drop-down Selection Mode menu to switch between modes.
```

Reviewer 2: What do you mean when you say "In this example, you can try out all the selection modes"? How can you try it out in an example?

Author: When you select a mode from the dropdown menu, the change is reflected in the List. However, to see the change, you have to try and select items. Please see the demo linked in the Trello card for examples

Reviewer 2: No, I mean why do you refer to a demo as an example? If you can switch selection modes in a demo, do not call it an example.

> List items can display checkboxes or radio buttons that help users select the items. To show these controls, enable the [showSelectionControls](/Documentation/ApiReference/UI_Components/dxList/Configuration/#showSelectionControls) property. Otherwise, users click or tap list items to select them. The **showSelectionControls** property must also be enabled when you use the *"all"* **selectionMode** so that the Select All checkbox is visible.

Reviewer 2: ```suggestion
List items can display checkboxes or radio buttons. To show these controls, enable the [showSelectionControls](/Documentation/ApiReference/UI_Components/dxList/Configuration/#showSelectionControls) property. Otherwise, users can click or tap list items to select them. The **showSelectionControls** property should also be enabled when you use the *"all"* **selectionMode** to show the Select All checkbox.
```

> When the **List** data is [paginated](/Documentation/Guide/UI_Components/List/Paging/), you can choose whether the Select All checkbox will select all items on all pages or only the items on the currently rendered page. To configure this functionality, set the [selectAllMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#selectAllMode) property to *"allPages"* or *"page"*. In this demo, you can use the Select All Mode drop-down menu to change the **selectAllMode** property at runtime.

Reviewer 2: ```suggestion
When the **List** data is [paginated](/Documentation/Guide/UI_Components/List/Paging/), you can choose whether the Select All checkbox selects all items on all pages or only the items on the current page. To configure this functionality, set the [selectAllMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#selectAllMode) property to *"allPages"* or *"page"*. In this demo, you can use the drop-down Select All Mode menu to change the **selectAllMode** property at runtime.
```

> When the **List** data is [paginated](/Documentation/Guide/UI_Components/List/Paging/), you can choose whether the Select All checkbox will select all items on all pages or only the items on the currently rendered pages. To configure this functionality, set the [selectAllMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#selectAllMode) property to *"allPages"* or *"page"*. In this demo, you can use the Select All Mode drop-down menu to change the **selectAllMode** property at runtime.

Reviewer 3: ```suggestion
When the **List** data is [paginated](/Documentation/Guide/UI_Components/List/Paging/), you can choose whether the Select All checkbox will select all items on all pages or only the items on the currently rendered page. To configure this functionality, set the [selectAllMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#selectAllMode) property to *"allPages"* or *"page"*. In this demo, you can use the Select All Mode drop-down menu to change the **selectAllMode** property at runtime.
```

> To configure the initial selection or access the keys of currently selected items, use the [selectedItemKeys](/Documentation/ApiReference/UI_Components/dxList/Configuration/#selectedItemKeys) array. In this example, you can see the contents of this array in the Selected IDs section below the List.

Reviewer 2: ```suggestion
To configure the initial selection or access the keys of selected items, use the [selectedItemKeys](/Documentation/ApiReference/UI_Components/dxList/Configuration/#selectedItemKeys) array. This example shows the contents of this array in the Selected IDs section below the List.
```

> Or use the *single* mode if you need to select only one item at a time. In this case, the selection is controlled with radio buttons.

Reviewer: You describe too much of the demo. Let's focus on the capabilities. I think you can take the [Selection - User Interaction](https://js.devexpress.com/Documentation/Guide/UI_Components/List/Selection/#User_Interaction) article, adapt it a bit to satisfy the guidelines, add information about `selectedItemKeys` and `onSelectionChanged` to the end, and you will have a decent description.

> You can also implement the [onSelectionChanged](/Documentation/ApiReference/UI_Components/dxList/Configuration/#onSelectionChanged) event handler to perform an action when users select an item or items.

Reviewer 2: ```suggestion
You can also implement the [onSelectionChanged](/Documentation/ApiReference/UI_Components/dxList/Configuration/#onSelectionChanged) event handler to perform an action when users select items.
```
